toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -201,12 +201,13 @@ func buildSRPMInChroot(chrootDir, rpmDirPath, toolchainDirPath, workerTar, srpmF
 	toolchainRpmsOverlayMount, toolchainRpmsOverlayExtraDirs := safechroot.NewOverlayMountPoint(chroot.RootDir(), overlaySource, chrootLocalToolchainDir, toolchainDirPath, chrootLocalToolchainDir, overlayWorkDirToolchain)
 	rpmCacheMount := safechroot.NewMountPoint(*cacheDir, chrootLocalRpmsCacheDir, "", safechroot.BindMountPointFlags, "")
 	mountPoints := []*safechroot.MountPoint{outRpmsOverlayMount, toolchainRpmsOverlayMount, rpmCacheMount}
+	extraDirs := append(outRpmsOverlayExtraDirs, chrootLocalRpmsCacheDir)
+	extraDirs = append(extraDirs, toolchainRpmsOverlayExtraDirs...)
 	if isCCacheEnabled(ccacheManager) {
 		ccacheMount := safechroot.NewMountPoint(ccacheManager.CurrentPkgGroup.CCacheDir, chrootCcacheDir, "", safechroot.BindMountPointFlags, "")
 		mountPoints = append(mountPoints, ccacheMount)
+		extraDirs = append(extraDirs, chrootCcacheDir)
```

Comment:
What is this needed for? (and why wasn't it needed before?) Could you please add a comment here explaining?

Reply:
It was actually needed before, I just moved where it's being done.
`extraDirs` are the directories to be created. Before this change, while I moved ccache logic under isCCacheEnabled(), I have left the directory creation outside. This change moves the creation of the ccache folder to also be under isCCacheEnabled().

Reply:
I'll add a comment explaining what extraDirs are...